### ISSUE TITLE
Parentheses not preserved in parse tree

### ISSUE THREAD

**Issue by mrfigg** [Morgul/rpgdice#5]
Parentheses are not preserved in the parse tree, which limits the ability to create a proper breakdown of the dice roll and results in `Expression.render()` returning an incorrect value. (For anyone reading this in the future, the broken math in this issue is being addressed in #4.)
![screenshot_2019-02-28](https://user-images.githubusercontent.com/36649520/53579702-e94a3600-3b2e-11e9-8dd1-9accd15c8527.png)
`Expression.render()` output: `3 - { 1d20: [ 8 ] } + 5 + 2`
```json
{
  "type": "subtract",
  "left": {
    "type": "number",
    "value": 3
  },
  "right": {
    "type": "add",
    "left": {
      "type": "add",
      "left": {
        "type": "roll",
        "count": 1,
        "sides": 20,
        "results": [
          8
        ],
        "value": 8
      },
      "right": {
        "type": "number",
        "value": 5
      },
      "value": 13
    },
    "right": {
      "type": "number",
      "value": 2
    },
    "value": 15
  },
  "value": -12
}
```
Maybe add a new `parentheses` `Expression` type?
```json
{
  "type": "parentheses",
  "content": {},
  "value": 1234
}
```

**Comment by Morgul** [comment]
Great catch. Yeah, adding a new Expression type for Parentheses would be required. And, in fact, might be contributing to the bad math; I am not convinced they even _do_ anything right now. (They might, but experimentally, they don't seem to.)

**Comment by Morgul** [comment]
closed by 6e6851d916cf809c390ecb2fda3ddd0e6cf323a6.
